rbac/api/utils.py

```diff
@@ -31,24 +31,33 @@ def populate_tenant_account_id():
 
 def get_resources(resource, org_id):
     """Get queryset by org_id."""
-    queryset = RESOURCE_MODEL_MAPPING[resource].objects.all()
+    model = RESOURCE_MODEL_MAPPING[resource]
+    queryset = model.objects.all()
     if org_id:
         if resource == "binding":
             raise ValueError("Binding cannot be filtered by org_id.")
         tenant = get_object_or_404(Tenant, org_id=org_id)
         queryset = queryset.filter(tenant=tenant)
-    return queryset
+    return queryset, model
 
 
 def migration_resource_deletion(resource, org_id):
     """Delete migration related resources."""
-    resource_objs = get_resources(resource, org_id)
+    resource_objs, model = get_resources(resource, org_id)
 
     if resource == "workspace":
         # Have to delete the ones without children first or deletion will fail
         logger.info("Deleting workspaces without children.")
-        while resource_objs.filter(children=None).exists():
-            resource_objs.filter(children=None).delete()
+        chunk_delete(resource_objs.filter(children=None), model)
         logger.info("All workspaces without children removed.")
-    resource_objs.delete()
+    chunk_delete(resource_objs, model)
     logger.info(f"Resources of type {resource} deleted.")
+
+
+def chunk_delete(queryset, model):
+    """Delete queryset in chunks."""
+    while True:
```

Comment:
Maybe something safer here would be to use [`queryset.exists()`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.exists)?
```suggestion
    while queryset.exists():
```